src/lava/proc/dense/process.py

```diff
@@ -8,16 +8,63 @@
 
 
 class Dense(AbstractProcess):
-    """Dense connections between neurons.
-    Realizes the following abstract behavior:
-    a_out = W * s_in
+    """Dense connections between neurons. Realizes the      following abstract
+    behavior: a_out = weights * s_in '
+
+    Parameters
+    ----------
+
+    weights:
+    Connection weight matrix.
+
+    weight_exp:
+    Shared weight exponent used to
+    scale magnitude of weights, if needed. Mostly for fixed point
+    implementations. Unnecessary for floating point implementations.
+    Default value is 0.
+
+    num_weight_bits:
+    Shared weight width/precision used by weight. Mostly for
+    fixed point  implementations. Unnecessary for floating point
+    implementations.
+    Default is for weights to use full 8 bit precision.
+
+    sign_mode:
+    Shared indicator whether synapse is of 'null' (0),
+    ’mixed’ (1, default), ’excitatory’ (2) or  ’inhibitory’ (3) type. If
+    ’mixed’, the sign of the weight is included in the weight bits and
+    the fixed point weight used for inference is scaled by 2.
+    Unnecessary for floating point implementations.
+
+    In the fixed point implementation, weights are scaled according to the
+    following equations:
+    weights = weights * (2 ** w_scale)
+    w_scale =  8 - num_weight_bits + weight_exp + isMixed()
+
+    a_buff:
+    Circular buffer that stores output activations accumulated in current
+    timestep for future timesteps.
+
     """
 
+    # ToDo: (DR) Implement a ProcModel that supports synaptic delays. a_buff
+    # must then be adjusted to the length of the delay.
+
+    # ToDo: (DR) Revisit the implementation of w_scale so that less of this
+    # computation is exposed at the level of the Process.
+
     def __init__(self, **kwargs):
-        # super(AbstractProcess, self).__init__(kwargs)
-        # shape = kwargs.pop("shape")
         super().__init__(**kwargs)
         shape = kwargs.get("shape", (1, 1))
+        weights = kwargs.pop("weights", 0)
```

Comment:
This is missing input validation. What if the user inputs a 4D tensor or a 1D vector - does that work? I would expect an exception to be raised immediately if I input data that is of the correct type but, for instance, wrong shape. Same is probably true for other inputs. Adding these would then require unit testing them - which is also missing at the moment.

This is too much to take on now for the release. Please create an issue for this in Github.

Reply:
I've added input validation + unit tests of that validation for the shape of 'shape' and 'weights', to ensure the user knows that Dense is expecting a 2D tensor. I will create an issue to add input validation for all other fields of this Process and any existing Processes in the Lava repo.